prime-router/src/main/kotlin/DocumentationFactory.kt

```diff
@@ -22,6 +22,11 @@ object DocumentationFactory {
         sb.appendLine("") // start with a blank line at the top 
         appendLabelAndData(sb, "Name", displayName)
         appendLabelAndData(sb, "Type", element.type?.name)
+		if (element.pii == true) {
```

Comment:
More idiomatic Kotlin would be something like:
     appendLabelAndData(sb, "PII", if (element.pii) "Yes" else "No")

The "if (...) ... else ... " expression in Kotlin is the equivalent of the ternary operation  "... ? ... : ..."  in java.

Reply:
Updated to something similar.  I had to add "== true".  I tend to prefer old syntax at times (unless it's too verbose) since it makes reading and rewriting it in a different language and data processing frameworks much easier.
I will have to read up more on it and will see how well Kotlin works with modern data related frameworks.

Reply:
One of the major benefits of Kotlin is that in many ways it can/does operate like a functional language, so you can tersely express your ideas in a type safe and null safe fashion.  

Another major benefit is that because it sits on top of the JVM, it works with any data framework that Java can, but you can often express things in a cleaner way. This is especially true with the fact that by default variables are immutable, which means you need to opt something in to mutation, and can know that when you pass around your variables they won't be changed, and there shouldn't be any side effects from calling a function (or very few. logging is an example where side effects can't be avoided)

One feature I particularly like is the notion of `let` in Kotlin. Whereas in Java or other languages where you need to do a null check before doing an action on it, in Kotlin you can say:

```kotlin
val returnVal: String? = someFunction()
returnVal?.let {
    // consume and do something with the variable, or, if it's null, don't. it will jus skip this check
}
```